core/src/main/scala/alleycats/ConsK.scala

```diff
@@ -0,0 +1,20 @@
+package alleycats
+
+import cats.SemigroupK
+import export.imports
+import simulacrum.typeclass
+
+@typeclass trait ConsK[F[_]] {
+  def cons[A](hd: A, tl: F[A]): F[A]
+}
+
+object ConsK extends ConsK0 {
+  implicit def pureSemigroupKIsConsK[F[_]](implicit p: Pure[F], s: SemigroupK[F]): ConsK[F] =
+    new ConsK[F] {
+      def cons[A](hd: A, tl: F[A]): F[A] = s.combine(p.pure(hd), tl)
+    }
+}
+
+@imports[ConsK]
+trait ConsK0
```

Comment:
Should this be `private[alleycats] sealed trait ConsK0`? I don't know if that causes any issues with export hook. But in general I think that if a trait just exists as an implementation detail then we shouldn't expose it for extension/usage by users of the library. If nothing else, making it private makes it obvious to users that they aren't meant to try to use it.